`make install` fails with `--prefix`, tries to install bash completions in system directories

Probably related to #7149, when I go to `make install` on dane (with no arguments other than `--prefix`, this fails because it tries to install the bash completions to `/usr/share`:

```
make[2]: Entering directory '/g/g0/hobbs17/flux-core/build/etc'
 /usr/bin/mkdir -p '/g/g0/hobbs17/flux-core/build/../install/libexec/flux'
 /usr/bin/mkdir -p '/g/g0/hobbs17/flux-core/build/../install/libexec/flux'
 /usr/bin/mkdir -p '/usr/share/bash-completion/completions'
 /usr/bin/mkdir -p '/g/g0/hobbs17/flux-core/build/../install/etc/flux/system/cron.d'
 /usr/bin/mkdir -p '/g/g0/hobbs17/flux-core/build/../install/share/flux/help.d'
 /usr/bin/mkdir -p '/g/g0/hobbs17/flux-core/build/../install/share/flux/.nodocs'
 /usr/bin/install -c completions/flux '/usr/share/bash-completion/completions'
 /usr/bin/install -c flux-run-system-scripts '/g/g0/hobbs17/flux-core/build/../install/libexec/flux'
/usr/bin/install: cannot remove '/usr/share/bash-completion/completions/flux' /usr/bin/install -c -m 644 ../../etc/kvs-backup.cron '/g/g0/hobbs17/flux-core/build/../install/etc/flux/system/cron.d'
: Permission denied
 /usr/bin/install -c -m 644 flux/.nodocs '/g/g0/hobbs17/flux-core/build/../install/share/flux/.nodocs'
 /usr/bin/install -c -m 644 flux/help.d/core.json '/g/g0/hobbs17/flux-core/build/../install/share/flux/help.d'
make[2]: *** [Makefile:558: install-bashcompSCRIPTS] Error 1
make[2]: *** Waiting for unfinished jobs....
 /usr/bin/mkdir -p '/g/g0/hobbs17/flux-core/build/../install/libexec/flux/modprobe/'
 /usr/bin/install -c ../../etc/modprobe/rc1.py ../../etc/modprobe/rc3.py '/g/g0/hobbs17/flux-core/build/../install/libexec/flux/modprobe/'
make[2]: Leaving directory '/g/g0/hobbs17/flux-core/build/etc'
make[1]: *** [Makefile:877: install-am] Error 2
make[1]: Leaving directory '/g/g0/hobbs17/flux-core/build/etc'
make: *** [Makefile:602: install-recursive] Error 1
```

--- a/configure.ac
+++ b/configure.ac
@@ -433,6 +433,7 @@ PKG_CHECK_MODULES([CURSES], [ncursesw], [], [])
 PKG_CHECK_MODULES([LIBARCHIVE], [libarchive], [], [])
 PKG_CHECK_MODULES([BASH_COMPLETION], [bash-completion], [
     bashcompdir="`$PKG_CONFIG \
+        --define-variable=prefix=${prefix} \
         --define-variable=datadir=${datadir} \
         --variable=completionsdir bash-completion`"
 ], [

